runhouse/resources/hardware/cluster_factory.py

```diff
@@ -138,6 +241,69 @@ def ondemand_cluster(
         >>> # Load cluster from above
         >>> reloaded_cluster = rh.ondemand_cluster(name="rh-4-a100s")
     """
+    if isinstance(server_connection_type, ServerConnectionType):
+        server_connection_type = server_connection_type.value
+
+    if server_connection_type in [AWS_SSM_CONN, PARAMIKO_CONN]:
+        raise ValueError(
+            f"OnDemandCluster does not support server connection type {server_connection_type}"
+        )
+
+    if not server_connection_type:
+        if open_ports or ssl_keyfile or ssl_certfile:
+            server_connection_type = HTTPS_CONN
+        elif server_host in Cluster.LOCAL_HOSTS:
+            server_connection_type = SSH_CONN
+        else:
+            server_connection_type = HTTP_CONN
+
+    if server_port is None:
+        if server_connection_type == HTTPS_CONN:
+            server_port = Cluster.DEFAULT_HTTPS_PORT
+        elif server_connection_type == HTTP_CONN:
+            server_port = Cluster.DEFAULT_HTTP_PORT
+        else:
+            server_port = Cluster.DEFAULT_SERVER_PORT
+
+    if (
+        server_connection_type in [HTTPS_CONN, HTTP_CONN]
+        and server_host in Cluster.LOCAL_HOSTS
+    ):
+        warnings.warn(
+            f"Server connection type set to {server_connection_type}, with server host set to {server_host}. "
+            f"Note that this will require opening an SSH tunnel to forward traffic from {server_host} to the server."
+        )
+
+    open_ports = (
+        []
+        if open_ports is None
+        else [open_ports]
+        if isinstance(open_ports, (int, str))
+        else open_ports
+    )
+
+    if open_ports:
```

Comment:
If the user has the server port in open_ports, is using HTTPS or HTTP, and den_auth disabled, we should raise a warning that the server is insecure, and must be inside a VPC or have den_auth enabled to secure it.